src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy

```diff
@@ -142,6 +145,12 @@ final class CheckerFrameworkPlugin implements Plugin<Project> {
       project.tasks.withType(AbstractCompile).all { compile ->
         if (compile.hasProperty('options') && (!userConfig.excludeTests || !compile.name.toLowerCase().contains("test"))) {
           compile.options.annotationProcessorPath = project.configurations.checkerFramework
+          if (javaVersion.java8 && new JarFile(project.configurations.checkerFramework.asPath)
```

Comment:
It might be a bit nicer to break out the logic to determine the CF version, either to a helper method or to code outside the condition.